### PR TITLE
RIA-8256 Added logic in the handler to put the Remittal related docum…

### PR DESCRIPTION
**Pull request by neha-aggarw**
…ent in an Object for the Documents tab
- Added new object to hold the documents (eq to complex type in the CCD).

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.RemittalDocument;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+
+@Service
+public class RemittalDocumentsAppender {
```
Comment (ThomasKKC) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java:12]: What is the difference between this and Appender<T>?

**Review by ThomasKKC — CHANGES_REQUESTED** [review]
Can we add a functional test for this?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        requireNonNull(existingRemittalDocuments, "existingDocuments must not be null");
+        requireNonNull(newRemittalDocument, "newRemittalDocument must not be null");
+
+        final List<IdValue<RemittalDocument>> allDocuments = new ArrayList<>();
+
+        int index = existingRemittalDocuments.size() + 1;
+
+        allDocuments.add(new IdValue<>(String.valueOf(index--), newRemittalDocument));
+
+        for (IdValue<RemittalDocument> existingDocument : existingRemittalDocuments) {
```
Comment (MiguelMartinRojas) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java:26]: What is the purpose of this prepend method? looks a little bit complicated to read

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java (hunk 1 above)
Comment (neha-aggarw) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java:12]: DocumentsAppender appends Documents & this class 'prepends' the RemittalDocument objects to an already existing collection.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java (hunk 2 above)
Comment (neha-aggarw) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/RemittalDocumentsAppender.java:26]: Prepend - to add in front of the existing collection. I can add a comment to explain the logic.

**Pull request closed without merging**